src/test_workflow/dependency_installer.py

```diff
@@ -25,24 +25,26 @@ def __init__(self, root_url, build_manifest, bundle_manifest):
         self.bundle_manifest = bundle_manifest
 
     def download_dist(self, dest):
-        local_path = os.path.join(dest, os.path.basename(self.bundle_manifest.build.location))
+        local_path = os.path.realpath(os.path.join(dest, os.path.basename(self.bundle_manifest.build.location)))
         return self.download_or_copy(self.bundle_manifest.build.location, local_path)
 
+    def __source_dest(self, path, category, dest):
+        source = "/".join([self.root_url, category, self.build_manifest.build.filename, path])
+        dest = os.path.realpath(os.path.join(dest, "/".join(path.split("/")[1:])))
+        return (source, dest)
+
     def download(self, paths, category, dest):
         logging.info(f"Downloading to {dest} ...")
         with concurrent.futures.ThreadPoolExecutor(max_workers=4) as executor:
-            for path in paths:
-                url = "/".join([self.root_url, category, path])
-                # paths are prefixed by category, remove
-                local_path = os.path.join(dest, "/".join(path.split("/")[1:]))
-                executor.submit(self.download_or_copy, url, local_path)
+            futures = executor.map(lambda args: self.download_or_copy(*args),
+                                   map(lambda path: self.__source_dest(path, category, dest), paths))
+            for future in concurrent.futures.as_completed(futures):
+                logging.info(future.result())
```

Comment:
Add a new unit test for capturing a failure result

Reply:
Done